addOns/graphql/graphql.gradle.kts

```diff
@@ -11,6 +11,9 @@ zapAddOn {
                 register("commonlib") {
                     version.set(">= 1.17.0 & < 2.0.0")
                 }
+                register("wappalyzer") {
```

Comment:
I don't see a reason for the dependency, the graphql add-on does not need the wappalyzer to work (or shouldn't need as is in the PR).

Reply:
I'll try removing it again and see if things behave, I had been having build issues.

Reply:
Just removing will not work, the code needs to be changed, this would have to be implemented with an optional extension to not have the hard dependency.

Reply:
Okay. I'll look at something that has custom payloads

Reply:
I'm now encountering:
```
89529 [ZAP-GraphQL-Parser1] ERROR org.zaproxy.zap.ZAP.UncaughtExceptionLogger - Exception in thread "ZAP-GraphQL-Parser1"
java.lang.NoClassDefFoundError: org/zaproxy/zap/extension/wappalyzer/ExtensionWappalyzer
	at org.zaproxy.addon.graphql.GraphQlFingerprinter.getExtTech(GraphQlFingerprinter.java:228) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlFingerprinter.fingerprint(GraphQlFingerprinter.java:107) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlParser.fingerprint(GraphQlParser.java:177) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlParser$2.run(GraphQlParser.java:165) ~[?:?]
Caused by: java.lang.ClassNotFoundException
	at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:330) ~[main/:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[?:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	... 4 more
```

when using this. (After making it an extension/optional dependency.)

Reply:
Now it's claiming the extension can't find the dependency:
```
2769497 [ZAP-GraphQL-Parser1] ERROR org.zaproxy.zap.ZAP.UncaughtExceptionLogger - Exception in thread "ZAP-GraphQL-Parser1"
java.lang.NoClassDefFoundError: org/zaproxy/zap/extension/wappalyzer/ExtensionWappalyzer
	at org.zaproxy.addon.graphql.techdetection.ExtensionTechDetection.<clinit>(ExtensionTechDetection.java:40) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlFingerprinter.fingerprint(GraphQlFingerprinter.java:100) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlParser.fingerprint(GraphQlParser.java:177) ~[?:?]
	at org.zaproxy.addon.graphql.GraphQlParser$2.run(GraphQlParser.java:165) ~[?:?]
Caused by: java.lang.ClassNotFoundException
	at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:330) ~[main/:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[?:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	... 4 more
```

Reply:
Same thing happens if I try to do it in hook instead of constructor.

Reply:
It's the optional extension that should mediate that, e.g. set something to `GraphQlFingerprinter` that would be called back.

Reply:
Sorted 😃